test/e2e/network_segmentation.go

```diff
@@ -62,314 +62,320 @@ var _ = Describe("Network Segmentation", func() {
-						append(ipCommand, "route", "add", "default", "via", gatewayIP.String(), "dev", "eth0")...)
-					Expect(err).NotTo(HaveOccurred())
-				}
-				// Now try to reach the host from the UDN pod
-				defaultPodHostIP := udnPod.Status.HostIPs
-				for _, hostIP := range defaultPodHostIP {
-					By("checking the UDN pod can't reach the host on IP " + hostIP.IP)
-					ping := "ping"
-					if utilnet.IsIPv6String(hostIP.IP) {
-						ping = "ping6"
-					}
-					Consistently(func() bool {
-						_, err := e2ekubectl.RunKubectl(udnPod.Namespace, "exec", udnPod.Name, "--",
-							ping, "-c", "1", "-W", "1", hostIP.IP,
-						)
-						return err == nil
-					}, 4*time.Second).Should(BeFalse())
-				}
-
-				By("asserting UDN pod can't reach default services via default network interface")
-				// route setup is already done, get kapi IPs
-				kapi, err := cs.CoreV1().Services("default").Get(context.Background(), "kubernetes", metav1.GetOptions{})
-				Expect(err).NotTo(HaveOccurred())
-				for _, kapiIP := range kapi.Spec.ClusterIPs {
-					By("checking the UDN pod can't reach kapi service on IP " + kapiIP)
-					Consistently(func() bool {
-						return connectToServerViaDefaultNetwork(udnPodConfig, kapiIP, int(kapi.Spec.Ports[0].Port)) != nil
-					}, 5*time.Second).Should(BeTrue())
-				}
 			},
-			Entry(
-				"with L2 dualstack primary UDN",
-				networkAttachmentConfigParams{
-					name:     nadName,
-					topology: "layer2",
-					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
-					role:     "primary",
-				},
-				*podConfig(
-					"udn-pod",
-					withCommand(func() []string {
-						return httpServerContainerCmd(port)
-					}),
-					withNodeSelector(map[string]string{nodeHostnameKey: workerOneNodeName}),
-				),
-			),
-			Entry(
-				"with L3 dualstack primary UDN",
-				networkAttachmentConfigParams{
-					name:     nadName,
-					topology: "layer3",
-					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
-					role:     "primary",
-				},
-				*podConfig(
-					"udn-pod",
-					withCommand(func() []string {
-						return httpServerContainerCmd(port)
-					}),
-					withNodeSelector(map[string]string{nodeHostnameKey: workerOneNodeName}),
-				),
-			),
+			Entry("plain network-attachment-definitions", func(c networkAttachmentConfigParams) error {
+				netConfig := newNetworkAttachmentConfig(c)
+				nad := generateNAD(netConfig)
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(context.Background(), nad, metav1.CreateOptions{})
+				return err
+			}),
+			Entry("user-defined-network CRD", func(c networkAttachmentConfigParams) error {
+				By(fmt.Sprintf("net params: %+v", c))
+				udnManifest := generateUserDefinedNetworkManifest(&c)
+				cleanup, err := createManifest(f.Namespace.Name, udnManifest)
+				DeferCleanup(cleanup)
+				Expect(waitForUserDefinedNetworkReady(f.Namespace.Name, c.name, 5*time.Second)).To(Succeed())
```

Comment:
would be nice to switch to api client here similar to NAD function

Reply:
Same here, it seems impossible to import OVN-K CRDs clients and types because the `go-controller/pkg/crd` is not exported in a way it can be consume from e2e tests or external project.
In addition I couldn't find similar example in e2e tests (using OVN-K CRD client)